File: app/src/main/java/com/xiaowu/myself/app/base/MyApplication.java
package com.xiaowu.myself.app.base;

import android.app.Application;
import android.content.Context;

public class MyApplication extends Application {
    private static MyApplication instance;
    private static Context context;

    public static MyApplication newInstance() {
        return instance;
    }

    /* 应用全局Context, 使用单例模式的业务类操作统一使用这个全局资源，避免传入Activity context,造成内存泄露 */
    public static Context getAppContext() {
        return context;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        instance = this;
        context = getApplicationContext();
    }
}

File: app/src/main/java/com/xiaowu/myself/app/base/MyApplication.kt
package com.xiaowu.myself.app.base

import android.app.Application
import android.content.Context
import android.content.SharedPreferences

class MyApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        mContext = this
        //初始化通用的SP&EDIT
        SP = getSharedPreferences("config", Context.MODE_PRIVATE)
        EDIT = SP?.edit()
    }

    companion object {
        var mContext: Context? = null
        /**
         * 初始化SP&EDIT
         */
        var SP: SharedPreferences? = null
        var EDIT: SharedPreferences.Editor? = null
    }
}
